pdns/pdns_recursor.cc

```diff
@@ -2463,17 +2471,29 @@ void distributeAsyncFunction(const string& packet, const pipefunc_t& func)
   tmsg->func = func;
   tmsg->wantAnswer = false;
 
-  if(write(tps.writeToThread, &tmsg, sizeof(tmsg)) != sizeof(tmsg)) {
+  ssize_t written = write(tps.writeQueriesToThread, &tmsg, sizeof(tmsg));
```

Comment:
Is there any specific reason you did not keep the test like it was before ? I find it more readable like that
```
  if (write(tps.writeQueriesToThread, &tmsg, sizeof(tmsg)) != sizeof(tmsg)) {                                                                                                 
    delete tmsg;                                                                                                                                                              
    int error = errno;                                                                                                                                                        
    if (error == EAGAIN || error == EWOULDBLOCK) {                                                                                                                            
      g_stats.queryPipeFullDrops++;                                                                                                                                           
    } else {                                                                                                                                                                  
      unixDie("write to thread pipe returned wrong size or error:" + error);                                                                                                  
    }                                                                                                                                                                         
  } 
```

Reply:
The value or `errno` only makes sense if `write()` returned `-1`, so we need to make sure it's the case, because otherwise we might be reading a garbage value of `errno` because of a short write. Agreed, it should not be happening because writing a 64-bits pointer on a pipe is supposed to be atomic since 64 < `PIPE_BUF`, but I'd rather be on the safe side.